osu.Game/IPC/LoadableIpcChannel.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Platform;
+
+#nullable enable
+
+namespace osu.Game.IPC
+{
+    public interface ILoadableIpcChannel
+    {
+        IpcMessage? HandleMessage(object message);
+    }
+
+    public abstract class LoadableIpcChannel<TMessage> : Component, ILoadableIpcChannel
```

Comment:
- does this need to be a component? that seems like a weird complication to everything
- what's the point of the interface if nothing really ever needs to use this (they could as well use the abstract class), and the interface is being actively [*cast out of*](https://github.com/ppy/osu/blob/71d17c21d1c8eba6eed6c7d8c3e88a66896a5cab/osu.Game/IPC/IpcServer.cs#L80-L81)?

Reply:
I wanted to use dependency injection to let the channels use what they need, but I guess passing only the game instance directly will be enough for now.